File: src/services/receipt/ReceiptApi/Models/Receipt.cs
namespace ReceiptApi.Models;

public class Receipt
{
    public int Id { get; set; }
    public string Message { get; set; } = null!;
    public int CustomerId { get; set; }
    public int TicketId { get; set; }
    public int CustomerTicketCount { get; set; }
    public DateTime CreatedAt { get; set; }

    public Receipt() {}

    public Receipt(int customerId, int ticketId, int customerTicketCount)
    {
        Message = "Receipt for generating a ticket";
        CustomerId = customerId;
        TicketId = ticketId;
        CustomerTicketCount = customerTicketCount;
        CreatedAt = DateTime.Now;
    }
}

File: src/clients/cli/src/receipt.rs
use serde::Deserialize;
use crate::{
    ticket::TicketData,
    customer::CustomerData
};

/// Holds receipt data.
#[derive(Deserialize)]
pub struct ReceiptData {
    /// Ticket id.
    pub id: i32,
    /// Ticket message.
    pub message: String,
    /// Customer id.
    #[serde(rename = "customerId")]
    pub customer_id: i32,
    /// Ticket id.
    #[serde(rename = "ticketId")]
    pub ticket_id: i32,
    /// Number of tickets created for the customer.
    #[serde(rename = "customerTicketCount")]
    pub customer_ticket_count: i32,
    /// Date and time when created in ISO-8601 format.
    #[serde(rename = "createdAt")]
    pub created_at: String,
}

/// Holds more data about receipt.
#[derive(Deserialize)]
pub struct ReceiptVerboseData {
    pub receipt: ReceiptData,
    pub ticket: TicketData,
    pub customer: CustomerData,
}
